space-cli/cmd/modules/accounts/set.go

```diff
@@ -0,0 +1,41 @@
+package accounts
+
+import (
+	"strings"
+
+	"github.com/AlecAivazis/survey/v2"
+	"github.com/spaceuptech/space-cli/cmd/utils"
+)
+
+func setAccount(prefix string) error {
+	credential, err := utils.GetCredentials()
+	if err != nil {
+		return err
+	}
+	exists := false
+	accountIDOptions := []string{}
+	for _, v := range credential.Accounts {
+		accountIDOptions = append(accountIDOptions, v.ID)
+		if prefix != "" {
+			if strings.HasPrefix(v.ID, prefix) {
+				prefix = v.ID
+				exists = true
+			}
+		}
+	}
+	if !exists {
+		if prefix != "" {
+			utils.LogInfo("Warning! No account found for prefix provided, showing all")
+		}
+		if err := survey.AskOne(&survey.Select{Message: "Choose the account ID to be set: ", Options: accountIDOptions}, &prefix); err != nil {
+			return err
+		}
+	}
+	credential.SelectedAccount = prefix
```

Comment:
Let the user choose which account he wants. Here you should show the filtered array.